clayer/pysertype.c

```diff
@@ -946,12 +783,53 @@ static ddspy_sertype_t *ddspy_sertype_new(PyObject *pytype)
     Py_XDECREF(pyname);
     Py_XDECREF(pykeyless);
     Py_XDECREF(pyversion_support);
-    Py_XDECREF(v0pykeysize);
-    Py_XDECREF(v2pykeysize);
 
     return new;
 }
 
+static dds_return_t init_cdrstream_descriptor(ddspy_sertype_t *sertype)
+{
+    dds_return_t ret = DDS_RETCODE_OK;
+
+    ddsi_typeinfo_t *type_info = ddsi_typeinfo_deser(sertype->typeinfo_ser_data, sertype->typeinfo_ser_sz);
+    if (type_info == NULL) {
+        ret = DDS_RETCODE_ERROR;
+        goto err_typeinfo;
+    }
+
+    const ddsi_typeid_t *type_id = ddsi_typeinfo_complete_typeid(type_info);
+    if (type_id == NULL) {
+        ret = DDS_RETCODE_ERROR;
+        goto err;
+    }
+
+    struct ddsi_domaingv * const gv = ddsrt_atomic_ldvoidp(&sertype->my_c_type.gv);
+    if (gv == NULL) {
+        ret = DDS_RETCODE_ERROR;
+        goto err;
+    }
+
+    const struct ddsi_type *ddsi_type = ddsi_type_lookup(gv, type_id);
+    if (ddsi_type == NULL) {
+        ret = DDS_RETCODE_ERROR;
+        goto err;
+    }
+
+    dds_topic_descriptor_t desc;
+    if ((ret = ddsi_topic_descriptor_from_type(gv, &desc, ddsi_type)) != DDS_RETCODE_OK)
+        goto err;
+
+    dds_cdrstream_desc_init(&sertype->cdrstream_desc, &cdrstream_allocator, desc.m_size, desc.m_align, desc.m_flagset, desc.m_ops, desc.m_keys, desc.m_nkeys);
+    sertype->v0_key_maxsize_bigger_16 = !(sertype->cdrstream_desc.flagset & DDS_TOPIC_FIXED_KEY);
+    sertype->v2_key_maxsize_bigger_16 = !(sertype->cdrstream_desc.flagset & DDS_TOPIC_FIXED_KEY_XCDR2);
+
```

Comment:
I think this leaks `desc`; and why not use `ddsi_typeinfo_free` for freeing `type_info`?